### PR TITLE
additional data processing options 

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

Added reclassify_links, use_direction_on_ways, and allow_alt_name as config options.  If `use_direction_on_ways = true` then use `direction` and `int_direction` on the way to update the directional for the `ref` and `int_ref`.

## Tasklist

 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/mjolnir/pbfgraphparser.cc [hunk 1]
```diff
@@ -1030,6 +1037,45 @@ struct graph_callback : public OSMPBF::Callback {
       }
     }
 
+    if (use_direction_on_ways_) {
```
Comment (mandeepsandhu) [inline src/mjolnir/pbfgraphparser.cc]: minor nit: the nested `if`s could be collapsed into:
```
if (use_direction_on_ways_ && !direction.empty() && !ref.empty())
```

File: src/mjolnir/pbfgraphparser.cc [hunk 2]
```diff
@@ -1030,6 +1037,45 @@ struct graph_callback : public OSMPBF::Callback {
...
+            for (uint32_t i = 0; i < refs.size(); i++) {
+              if (!refs.at(i).empty() && !directions.at(i).empty())
+                w.set_ref_index(osmdata_.name_offset_map.index(refs.at(i) + " " + directions.at(i)));
+            }
+          } else w.set_ref_index(osmdata_.name_offset_map.index(ref));
+        }
+      }
+    }
+
+    if (use_direction_on_ways_) {
```
Comment (mandeepsandhu) [inline src/mjolnir/pbfgraphparser.cc]: Same comment as before, these could be collapsed into a single stmt (less verbose).

File: src/mjolnir/pbfgraphparser.cc [hunk 3]
```diff
@@ -65,6 +65,8 @@ struct graph_callback : public OSMPBF::Callback {
     infer_internal_intersections_ =
         pt.get<bool>("data_processing.infer_internal_intersections", true);
     infer_turn_channels_ = pt.get<bool>("data_processing.infer_turn_channels", true);
+    use_direction_on_ways_ = pt.get<bool>("data_processing.use_direction_on_ways", false);
```
Comment (dgearhart) [inline src/mjolnir/pbfgraphparser.cc]: where does `reclassify_links` get set?

File: src/mjolnir/pbfgraphparser.cc [hunk 4]
```diff
@@ -1030,6 +1037,74 @@ struct graph_callback : public OSMPBF::Callback {
       }
     }
 
+    if (use_direction_on_ways_ && !ref.empty()) {
```
Comment (dgearhart) [inline src/mjolnir/pbfgraphparser.cc]: do we have any tests for the direction assignments?

File: scripts/valhalla_build_config [hunk 5]
```diff
@@ -268,11 +271,14 @@ help_text = {
...
     'data_processing': {
       'infer_internal_intersections': 'bool indicating whether or not to infer internal intersections during the graph enhancer phase or use the internal_intersection key from the pbf',
       'infer_turn_channels': 'bool indicating whether or not to infer turn channels during the graph enhancer phase or use the turn_channel key from the pbf',
       'apply_country_overrides': 'bool indicating whether or not to apply country overrides during the graph enhancer phase',
-      'use_admin_db': 'bool indicating whether or not to use the administrative database during the graph enhancer phase or use the admin keys from the pbf that are set on the way'
+      'use_admin_db': 'bool indicating whether or not to use the administrative database during the graph enhancer phase or use the admin keys from the pbf that are set on the way',
+      'use_direction_on_ways': 'bool indicating whether or not to process the direction key on the ways or utilize the guidance relation tags during the parsing phase',
+      'allow_alt_name': 'bool indicating whether or not to process the alt_name key on the ways during the parsing phase'
```
Comment (dgearhart) [inline scripts/valhalla_build_config]: in the future might want to use the `name:<lg>` for street signs with multiple languages

Comment (gknisely) [reply scripts/valhalla_build_config]: Agree

File: src/mjolnir/pbfgraphparser.cc (hunk 3 above)
Comment (gknisely) [reply src/mjolnir/pbfgraphparser.cc]: https://github.com/valhalla/valhalla/blob/2fe491eea546eef74cd82fd1140a4689d33a61bb/src/mjolnir/graphbuilder.cc#L1168

After i enabled this I noticed that I want to move and parser, builder, or enhancer flags to under the `data_processing` section in the config.

File: src/mjolnir/pbfgraphparser.cc (hunk 4 above)
Comment (gknisely) [reply src/mjolnir/pbfgraphparser.cc]: will add some

**Pull request merged**